core/src/test/java/kafka/server/BrokerRegistrationRequestTest.java

```diff
@@ -0,0 +1,217 @@
+                .setMaxSupportedVersion(feature.latestTesting())));
+
+        var listener = new BrokerRegistrationRequestData.Listener()
+                .setName("EXTERNAL")
+                .setHost("example.com")
+                .setPort(8082)
+                .setSecurityProtocol(SecurityProtocol.PLAINTEXT.id);
+        
+        var req = new BrokerRegistrationRequestData()
+            .setBrokerId(brokerId)
+            .setLogDirs(List.of(Uuid.randomUuid()))
+            .setClusterId(clusterId)
+            .setIncarnationId(Uuid.randomUuid())
+            .setIsMigratingZkBroker(zkEpoch != null)
+            .setFeatures(features)
+            .setListeners(new BrokerRegistrationRequestData.ListenerCollection(List.of(listener).iterator()));
+
+        BrokerRegistrationResponse resp = this.sendAndReceive(
+            channelManager, 
+            new BrokerRegistrationRequest.Builder(req), 
+            30000
+        );
+        return Errors.forCode(resp.data().errorCode());
+    }
+
+    @ClusterTest(types = {Type.KRAFT}, controllers = 1, metadataVersion = MetadataVersion.IBP_3_3_IV3)
+    public void testRegisterZkWith33Controller(ClusterInstance clusterInstance) throws Exception {
+        // Verify that a controller running an old metadata.version cannot register a ZK broker
+        var clusterId = clusterInstance.clusterId();
+        var channelManager = brokerToControllerChannelManager(clusterInstance);
+        try {
+            channelManager.start();
+            // Invalid registration (isMigratingZkBroker, but MV does not support migrations)
+            Assertions.assertEquals(
+                Errors.BROKER_ID_NOT_REGISTERED,
+                registerBroker(channelManager, clusterId, 100, 1L, 
+                    new FeatureLevel(MetadataVersionTestUtils.IBP_3_3_IV0_FEATURE_LEVEL, MetadataVersionTestUtils.IBP_3_3_IV3_FEATURE_LEVEL))
+            );
+
+            // No features (MV) sent with registration, controller can't verify
+            Assertions.assertEquals(
+                Errors.INVALID_REGISTRATION,
+                registerBroker(channelManager, clusterId, 100, null, null)
+            );
+
+            // Given MV is too high for controller to support
+            Assertions.assertEquals(
+                Errors.UNSUPPORTED_VERSION,
+                registerBroker(channelManager, clusterId, 100, null, 
+                    new FeatureLevel(MetadataVersionTestUtils.IBP_3_4_IV0_FEATURE_LEVEL, MetadataVersionTestUtils.IBP_3_4_IV0_FEATURE_LEVEL))
+            );
+
+            // Controller supports this MV and isMigratingZkBroker is false, so this one works
+            Assertions.assertEquals(
+                Errors.NONE,
+                registerBroker(channelManager, clusterId, 100, null, 
+                    new FeatureLevel(MetadataVersionTestUtils.IBP_3_3_IV3_FEATURE_LEVEL, MetadataVersionTestUtils.IBP_3_4_IV0_FEATURE_LEVEL))
```

Comment:
`MetadataVersionTestUtils.IBP_3_3_IV3_FEATURE_LEVEL` -> `MetadataVersion.IBP_3_3_IV3.featureLevel()`

Reply:
ditto to `IBP_3_4_IV0_FEATURE_LEVEL`